nix/packages/repline.nix

```diff
@@ -1,13 +1,20 @@
-{ mkDerivation, base, containers, fail, haskeline, mtl, process
-, stdenv
+{ mkDerivation, base, containers, exceptions, fetchgit, haskeline
+, mtl, process, stdenv
 }:
 mkDerivation {
   pname = "repline";
-  version = "0.2.1.0";
-  sha256 = "45c3186ff35ed650ee9c641f545a30860eedc44107fefb21da36df47aeb1ae7b";
+  version = "0.4.0.0";
+  src = fetchgit {
+    url = "https://github.com/basile-henry/repline.git";
+    sha256 = "0b72aagm51sr8dv86wcidhmr2yi2imdk5dxyk9dgarflvfy7vyw5";
+    rev = "cfdbb815d4410db3bbebae3a53fd03328de50d48";
+    fetchSubmodules = true;
+  };
   libraryHaskellDepends = [
-    base containers fail haskeline mtl process
+    base containers exceptions haskeline mtl process
   ];
+  testHaskellDepends = [ base containers mtl process ];
+  doHaddock = false;
```

Comment:
I think we're hitting a weird GHC 8.4.4 haddock bug:

```
Haddock coverage:

src/System/Console/Repline.hs:266:3: error:
    parse error on input ‘-- | banner function’
    |
266 |   -- | banner function
    |   ^^^^^^^^^^^^^^^^^^^^
```

At https://github.com/sdiehl/repline/blob/cfdbb815d4410db3bbebae3a53fd03328de50d48/src/System/Console/Repline.hs#L263-L267

This does not happen with GHC 8.6.5!

Reply:
Looks like something that would be good to fix in time for 0.4!

Also good to check in CI…

Reply:
It's broken again, it's originally an `ormolu` issue and it is marked as _wontfix_: https://github.com/tweag/ormolu/issues/624